pkg/api/index.go

```diff
@@ -109,8 +110,14 @@ func (hs *HTTPServer) getAppLinks(c *models.ReqContext) ([]*dtos.NavLink, error)
 			}
 
 			if include.Type == "dashboard" && include.AddToNav {
+				uid := include.UID
```

Comment:
You'll need the same logic applied for creating the `appLink` above, but `DefaultNavUrl` is set in the plugin manager code somewhere.